### DISCUSSION TITLE
Force signature method ed25519ph for ed25519 key

### DISCUSSION THREAD

**Discussion by commel** [openssl/openssl#23387] [Q&A] [answered]
This is on OpenSSL 3.2 and I am generating an ed25519 key. To sign some payload I am using ed25591ph, which I can configure using 
```
openssl pkeyutl -sign -inkey key.pem  -rawin -in msg.txt -pkeyopt instance:ed25591ph -out msg.txt.sig
```

When someone wants to verify the signature against my public key this works:

```
openssl pkeyutl -verify -rawin -pubin -inkey key.pub -pkeyopt instance:ed25519ph -in msg.txt -sigfile msg.txt.sig
```

The problem here is that the other side needs to know to explicitly use ```ed25519ph```. Is there a way to annotate this somehow? Maybe in the public key using some annotations?

**Comment by t8m** [comment]
The pkeyutl tool creates raw signatures without metadata. There is unfortunately no way to associate any metadata with this format. You will have to wrap the signature in some metadata file or hardcode the algorithm.

**Reply by commel** [reply]
So the way would be to extend ```pkeyutl -sign``` to output some signature container format, maybe like [ASiC?](https://en.wikipedia.org/wiki/Associated_Signature_Containers) And enable ```-verify``` to read this container format?

**Reply by t8m** [reply]
Yeah, that could be a possible new feature.

**Reply by t8m** [reply, marked as answer]
But BTW we have already the cms utility which is supposed to provide such kind of format.

**Comment by maborg** [comment]
No future for **ed25591ph**  in openssl CLI? 
Seem to be still not supported, would be nice to have a simple way to verify ed25591ph signature in a script.
